File: leetcode/java/832. Flipping an Image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        final int rows = image.length;
        final int cols = image[0].length;
        
        final int[][] ret = new int[rows][cols];
        final boolean isEvenLen = rows % 2 == 0;
        
        for(int i = 0; i < rows; i++) {    
            
            for(int j = 0; j < cols / 2; j++) {
                ret[i][j] = image[i][cols - 1 - j];
                ret[i][j] = ret[i][j] == 1 ? 0 : 1;
            }
            
            for(int j = (isEvenLen) ? cols / 2 : cols / 2 + 1; j < cols; j++) {
                ret[i][j] = image[i][cols - 1 - j];
                ret[i][j] = ret[i][j] == 1 ? 0 : 1;
            }
            
            if(!isEvenLen) {
                ret[i][cols / 2] = image[i][cols / 2];
                ret[i][cols / 2] = ret[i][cols / 2] == 1 ? 0 : 1;
            }
        }
        
        return ret;
    }
}

File: leetcode/kotlin/832. Flipping an Image.kt
class Solution {
        
    fun flipAndInvertImage(image: Array<IntArray>): Array<IntArray> {
        val rows = image.size;
        val cols = image[0].size;
        
        // 2차원 배열
        val ret = Array(rows) { IntArray(cols) {0} }
    
        for(i in 0 until rows) {
            
            for(j in 0 until cols / 2) {
                ret[i][j] = image[i][cols - 1 - j]
                ret[i][j] = if(ret[i][j] == 1) 0 else 1
            }
            
            val start = if(cols % 2 == 0) (cols / 2) else (cols / 2 + 1)
            for(j in start until cols) {
                ret[i][j] = image[i][cols - 1 - j]
                ret[i][j] = if(ret[i][j] == 1) 0 else 1
            }
            
            if(cols % 2 != 0) {
                ret[i][cols / 2] = image[i][cols / 2]
                ret[i][cols / 2] = if(ret[i][cols / 2] == 1) 0 else 1
            }
        }
    
        return ret;
    }
}
